### ISSUE TITLE
[App] Cli monitoring tool

### ISSUE THREAD

**Issue by lpgeiger** [OLSF/libra#398]
Feature branch: `moonshot-cli`

# Goal
Provide Statistics to the node operator from command line. 
- [x] provide chain metadata from upstream fullnode: height, epoch, validator set count.
- [x] provide state of the local node's sync progress
- [x] compare local node to a reference upstream fullnode
- [x] provide account information of the validator owner account

# Features

Runs a snapshot, or continuously in `live` mode

# UX
```
# single snapshot is default
ol-cli monitor

# continous live mode
ol-cli monitor --live
```

# Metrics & Output

Chain info:
- [x] Epoch (upstream): Int
- [x] Height (upstream): Int
- [x] Validator count: Int
- [x] Latest Waypoint: Waypoint

Node Info:
- [x] Node Operating: Bool
- [x] Is mining: Bool
- [x] Is synced: Bool
- [ ] Epoch (local): Int
- [x] Height (local): Int
- [ ] Latest block signed: Int

Account Info:
- [x] Account: Hex
- [x] Exists on Chain: Bool
- [x] Is in Validator set: Bool
- [ ] Is Jailed: Bool
- [x] Tower Height (on chain):  Int
- [ ] Tower Height (local): Int
- [x] Latest epoch mining (on chain): Int

# Proposed architecture

- Abscissa for command line app
- Sled for a lightweight rust native local database
- Clokwerk for timers for live monitoring

**Comment by sm86** [comment]
Validator info [If in validator set]
- Number of blocks signed in last epoch
- Number of blocks proposed in last epoch
- Last voted block

**Comment by zmanian** [comment]
The Iqlusion 0L development program offers 400,000 GAS bounty for the completion of this issues.

We except to allocate 300,000 GAS for code and 100,000 GAS for code review.

**Comment by liangping** [comment]
I would like add [this library](https://github.com/fdehau/tui-rs) for this feature.

**Comment by earthwindfirewater** [comment]
closed by #425
